service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapperTest.java

```diff
@@ -233,4 +241,14 @@ public void When_MappingParsedConditionWithNonExistentReferenceInRelatedClinical
             .isExactlyInstanceOf(EhrMapperException.class)
             .hasMessage("Could not resolve Condition Related Medical Content reference");
     }
+
+    @Test
+    public void When_MappingParsedConditionCodeIsMissing_Expect_MapperException() throws IOException {
+        var jsonInput = ResourceTestFileUtils.getFileContent(INPUT_JSON_MISSING_CONDITION_CODE);
+        Condition parsedObservation = fhirParseService.parseResource(jsonInput, Condition.class);
+
+        assumeThatThrownBy(() -> conditionLinkSetMapper.mapConditionToLinkSet(parsedObservation, false))
```

Comment:
There's no need to assume here, we can use assertions instead
```suggestion
        assertThatThrownBy(() -> conditionLinkSetMapper.mapConditionToLinkSet(parsedObservation, false))
```